__e2e__/commands/fetchtask.test.js

```diff
@@ -0,0 +1,136 @@
+'use strict';
+
+const path = require('path');
+const execa = require('execa');
+const test = require('ava');
+const fs = require('fs');
+
+const rootCommand = path.join(process.cwd(), 'bin/index.js');
+const configFilePath = path.join(process.cwd(), 'config.json');
+
+// learning track python, javascript
+const learningTracksInfo = [
+  { trackName: 'Python', fileExtension: 'py' },
+  { trackName: 'JavaScript', fileExtension: 'js' },
+];
+
+const userDefaultConfig = {
+  learningTrack: '',
+  userName: 'testConfig',
+  taskCount: 0,
+  keys: [],
+  userSubmittedFiles: [],
+};
+
+const createUserConfig = (
+  trackName,
+  fileExtension,
+  keysNumber,
+  filesNumber,
+) => {
+  let userConfig = userDefaultConfig;
+  userConfig['learningTrack'] = trackName;
+  for (let i = 0; i < keysNumber; i++)
+    userConfig['keys'].push('testKey' + (i + 1));
+  for (let i = 0; i < filesNumber; i++)
+    userConfig['userSubmittedFiles'].push(
+      'task' + (i + 1) + '.' + fileExtension,
+    );
+  userConfig['taskCount'] = filesNumber;
+
+  return userConfig;
+};
+
+test.serial('no config file for fetchtask', async t => {
+  if (fs.existsSync(configFilePath)) fs.unlinkSync(configFilePath);
+  const { stdout } = await execa(rootCommand, ['fetchtask'], {
+    reject: false,
+  });
+  t.snapshot(stdout);
+});
+
+test.skip('incorrect key for fetchtask', async t => {
+  let snap = null;
+  t.assert(learningTracksInfo.length > 0);
+  for (let index in learningTracksInfo) {
+    const learningTrack = learningTracksInfo[index];
+    const trackName = learningTrack['trackName'];
+    const fileExtension = learningTrack['fileExtension'];
+    const userConfig = createUserConfig(trackName, fileExtension, 30, 30);
+    fs.writeFileSync(configFilePath, JSON.stringify(userConfig));
+    const { stdout } = await execa(rootCommand, ['fetchtask', 'testKey0']);
+    fs.unlinkSync(configFilePath);
+    if (snap) t.true(snap === stdout);
+    snap = stdout;
+  }
+  t.snapshot(snap);
+});
+
+test.serial('display completed task with fetchtask key', async t => {
+  t.assert(learningTracksInfo.length > 0);
+  for (let index = 0; index < learningTracksInfo; index += 1) {
+    const learningTrack = learningTracksInfo[index];
+    const trackName = learningTrack['trackName'];
+    const fileExtension = learningTrack['fileExtension'];
+    let userConfig = createUserConfig(trackName, fileExtension, 6, 5);
+    fs.writeFileSync(configFilePath, JSON.stringify(userConfig));
+    const { stdout } = await execa(rootCommand, ['fetchtask', 'testKey2'], {
+      reject: false,
+    });
```

Comment:
`Promise` isn't rejected here, right?

Reply:
It should not be rejected but right now the program is rejecting them, that's why I kept it right now to see where the flow is going, it will be removed eventually when the program is corrected.

Reply:
Or should I write the correct tests first and then correct the program accordingly as it should be done in test driven development.